x/ccv/provider/keeper/relay.go

```diff
@@ -55,11 +55,19 @@ func (k Keeper) OnRecvVSCMaturedPacket(
 // bypassing HandleThrottleQueues.
 func (k Keeper) HandleLeadingVSCMaturedPackets(ctx sdk.Context) {
 	for _, chain := range k.GetAllConsumerChains(ctx) {
+		// Note: it's assumed the order of the vsc matured slice matches the order of the ibc seq nums slice,
+		// in that a vsc matured packet data at index i is associated with the ibc seq num at index i.
 		leadingVscMatured, ibcSeqNums := k.GetLeadingVSCMaturedData(ctx, chain.ChainId)
-		for _, data := range leadingVscMatured {
+		ibcSeqNumsHandled := []uint64{}
+		for idx, data := range leadingVscMatured {
+			if k.VSCMaturedHandledLimitReached(ctx) {
+				// Break from inner for-loop, DeleteThrottledPacketData will still be called for this consumer
+				break
+			}
 			k.HandleVSCMaturedPacket(ctx, chain.ChainId, data)
+			ibcSeqNumsHandled = append(ibcSeqNumsHandled, ibcSeqNums[idx])
 		}
-		k.DeleteThrottledPacketData(ctx, chain.ChainId, ibcSeqNums...)
+		k.DeleteThrottledPacketData(ctx, chain.ChainId, ibcSeqNumsHandled...)
```

Comment:
Not really related to this PR, but why do we need the IBC packet seq number to store VSCMaturedPackets? We could just use the vscID, right?

Reply:
We need ibc seq and not vscID because you could have a slash packet and vsc matured packet that would overwrite one another if they are relevant to the same vscID. ibcSeq is always unique